### ISSUE TITLE
SC2120: don't flag if referenced argument has fallback

### ISSUE THREAD

**Issue by emsaks** [koalaman/shellcheck#1674]
#### For bugs
- Rule Id SC2120
- version .6 + online

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /bin/bash
function f () { echo "arg 1 is: ${1:-fallback}";}

f

```

#### Here's what shellcheck currently says:

^-- SC2120: f references arguments, but none are ever passed.

#### Here's what I wanted or expected to see:

Nothing

Since the function provides it's own fallback, obviously parameters are optional.
